src/main/webapp/WEB-INF/user.jsp

```diff
@@ -41,11 +42,23 @@
     </form>
     <hr/>
 
+    <% if(isUserLoggedIn){ %>
+    <form id="message-form" action="/messages?recipient=<%= request.getAttribute("user") %>" method="POST" class>
+    <% } else { %>
+    <form id="message-form" action="/messages?recipient=" method="POST" class="hidden">
```

Comment:
How about just putting the whole form in the "if" section instead of just the opening tag? There isn't any way to make this unhidden w/o a new page load right? If that is true then I think it is better to just not send the form in the first place. This could also apply to the about-me form above.